### ISSUE TITLE
Icon reloading logic problem

### ISSUE THREAD

**Issue by maxchang3** [natemoo-re/astro-icon#257]
### What version of `astro-icon` are you using?

v1.1.4

### Astro Info

```block
Astro                    v4.14.2
Node                     v23.1.0
System                   macOS (arm64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             astro-icon
```

### Describe the Bug

The current file-watching logic (by #251) does not apply additional filters and triggers unnecessary reloads when files inside folders, such as `src/` or `.astro/`(added to the watcher by others) are modified. This is not the expected behavior. 

Additionally, the watcher only listens for `change` events, meaning that if an SVG file is added to the `iconDir`, it won't trigger a reload (#207). 

This behavior is partly masked because users are likely to edit code that involves the `<Icon />` component in `.astro` files after adding an SVG file, which coincidentally triggers a reload due to the lack of filtering in the change event listener.

![image](https://github.com/user-attachments/assets/a2267d61-c676-4548-98c6-e0b7c845496d)

### What's the expected result?

+ Only relevant file changes should trigger the icon reload:
    - Changes in `.svg` files inside the `iconDir` directory should trigger the reload.
    - Changes in the `astro.config` file should also trigger the reload.
+ New SVG files added to `iconDir` should trigger a reload:
    - The reload logic should trigger when new SVG files are added to the `iconDir`, not just when existing files are changed.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-vqn7dv-26dhmj

### Related
#207
#251

**Comment by madcampos** [comment]
I think this may relate to an issue where changing a CSS file don't trigger the live reload, so the project never picks up the CSS changes.

**Comment by stramel** [comment]
Sorry everyone, I'll take a look today and work on getting this released

**Comment by SamuNatsu** [comment]
Still not fixed?

**Comment by stramel** [comment]
Fixed in v1.1.5
